README.md

```diff
@@ -1,97 +1,30 @@
-# Snap Plugin Library for Go
+## Snap Plugin Library for Go
 
-This is a library for writing plugins for the [snap](https://github.com/intelsdi-x/snap) telemetry framework. The goal for this library is to make it super simple to write a plugin.
+This is a library for writing plugins in Go for the [Snap telemetry framework](https://github.com/intelsdi-x/snap). 
 
-# Writing a plugin
+Snap has three different plugin types and for instructions on how to write a plugin check out the [collector](examples/collector/README.md), [processor](examples/processor/README.md), and [publisher](examples/publisher/README.md) plugin docs.
 
-In order to write a plugin for snap, it is necessary to define some methods to satisfy the appropriate interfaces. The interface is slightly different depending on what type (collector, processor, or publisher) of plugin is being written. The interfaces are:
+Before writing a snap plugin:
 
+* Verify it doesn't already exist by reviewing the [Plugin Catalog](https://github.com/intelsdi-x/snap/blob/master/docs/PLUGIN_CATALOG.md) 
```

Comment:
minor request, but thinking this could be less formal: 

Before writing a Snap plugin: 

* See if one already exists in the [Plugin Catalog]
* See if someone mentioned it in [the plugin wishlist]